components/brave_wallet/browser/nft_metadata_fetcher.cc

```diff
@@ -24,24 +26,15 @@
 
 namespace {
 
-std::optional<uint32_t> DecodeUint32(const std::vector<uint8_t>& input,
+std::optional<uint32_t> DecodeUint32(base::span<const uint8_t> input,
                                      size_t& offset) {
-  if (offset >= input.size() || input.size() - offset < sizeof(uint32_t)) {
+  if (input.size() < offset + sizeof(uint32_t)) {
```

Comment:
Strictly speaking this might overflow now.

Reply:
Reverted.